### ISSUE TITLE
Request: provide logging for phases before iterations start

### ISSUE THREAD

**Issue by cciccole** [CannyLab/tsne-cuda#40]
It would be useful if there were some sort of logging during the initialization, KNN search, etc., phases before the iterations start. With large datasets (e.g., millions of rows) there can be significant lag times with no reporting. Then it becomes a question whether any progress is actually being made.

On that note, do you have any theoretical or practical guidance on the maximum number of rows and/or columns that the algorithm can handle?

**Comment by DavidMChan** [comment]
We can look at adding some more logging in the next release. I think a lot of the code is there to handle it. We'd welcome a PR adding some logging as well! 

In theory, the number of rows/columns is limited by the GPU memory of a single GPU on your machine. With our Titan X GPUs (12GB of RAM), we were able to handle 10,000,000 rows in 50 columns in under 10 minutes. The limits we were seeing with the barnes-hut variant were between 10-15M points in 50-100K dimensions, however we haven't run detailed testing at the higher end of the number of points for the most recent version of the code.

When you add more points, it makes sense to enable multi-gpu processing for efficiency in the KNN computation, which we have experimented with, but haven't added to the latest release as a feature.
